docs/intro/tutorial.rst

```diff
@@ -238,6 +238,182 @@ At the end of our job, the spider should look like this:
+configure different Page Objects for different sites, because the extraction logic
+surely changes from site to site.
+This is exactly the functionality that *overrides* provides.
+
+Note that the crawling logic of the ``BooksSpider`` is pretty simple and straightforward:
+
+1. Extract all books URLs from the listing page
+2. For each book URL found in the step 1, fetch the page and extract the resultant item
+
+This logic should work without any change for different books sites because
+having pages with lists of books and then detail pages with the individual book is
+such a common way of structuring sites.
+
+Let's refactor the spider presented in the former section so that it also supports
+extracting books from the page `bookpage.com/reviews <https://bookpage.com/reviews>`_
+as well.
+
+The steps to follow are:
+
+#. Make our spider generic: move the remaining extraction code from the spider to a Page Object
+#. Configure *overrides* for Books to Scrape
+#. Add support for another site (Book Page site)
+
+Making the spider generic
+-------------------------
+This is almost done. The book extraction logic has been already moved to the
+``BookPage`` Page Object, but extraction logic to obtain the list of URL to books
+is already present in the ``parse`` method. It must be moved to its own Page
+Object:
+
+.. code-block:: python
+
+    class BookListPage(WebPage):
+
+        def book_urls(self):
+            return self.css('.article-info a::attr(href)').getall()
+
+Let's adapt the spider to use this new Page Object:
+
+.. code-block:: python
+
+    class BooksSpider(scrapy.Spider):
+        name = 'books_spider'
+        start_urls = ['http://books.toscrape.com/']
+        parse_book = callback_for(BookPage)  # extract items from book pages
+
+        def parse(self, response, page: BookListPage):
+            yield from response.follow_all(page.books_urls(), self.parse_book)
+
+All the extraction logic that is specific to the site is now responsibility
+of the Page Objects. As a result, the spider is now *site-agnostic* and will
+work providing that the Page Objects do their work.
+
+In fact, the spider only responsibility becomes expressing the crawling strategy:
+"fetch a list of item URLs, follow them, and extract the resultant items".
+The code gets clearer and simpler.
+
+Configure *overrides* for Books to Scrape
+-----------------------------------------
+It is convenient to create bases classes for the Page Objects given that we are going
+to have several implementations of the same Page Object (one per each site).
+The following code snippet introduces such base classes and refactors the
+existing Page Objects as subclasses of them:
+
+.. code-block:: python
+
+    # ------ Base page objects ------
+
+    class BookListPage(WebPage):
+
+        def book_urls(self):
+            return []
+
+
+    class BookPage(ItemWebPage):
+
+        def to_item(self):
+            return {}
```

Comment:
This is a reasonable implementation, but if I'd need to write this cod efor non-educational purposes I'd probably use abc for BookListPage, and remove to_item method from BookPage (as it is already defined, and it is an abstract method in the base class). 

That's a nitpick, and doing as per ^^ suggestion could make code less clear. Feel free to decide what to do @ivanprado :)

Reply:
If the methods were abstract, the spider would raise non implemented errors for URLs not belonging to the domain. I thought it would be better to have a default implementation that just does nothing in such a case. Yeah, but I'm unsure about that. Some users might prefer to see a failure because it could be a failure in the crawling logic. Let's keep it as is for clarity reasons.

However, I've noticed that is better to change the default `to_item()` to return `None`. Otherwise empty items would be returned if the callback is used for a non overridden domain. Done here: https://github.com/scrapinghub/scrapy-poet/pull/43/commits/cbb7e5cc60b30d8cbd79e1edc7144acb4efe1818